lib/action_controller/serialization.rb

```diff
@@ -8,8 +8,10 @@ module Serialization
     include ActionController::Renderers
 
     module ClassMethods
-      def serialization_scope(scope)
-        self._serialization_scope = scope
+      def serialization_scope(scope, opts = {})
+        before_action(opts.slice(:only, :except, :if, :unless)) do
```

Comment:
I also would except there to be bugs with `current_user` being called in a `before_action` with all the side-effects authentication-related libs can have